### PR TITLE
libucl: new recipe

### PR DESCRIPTION
**Pull request by hoanga**
new recipe for libucl

Libucl is a parser and C API to parse and generate ucl objects.

### CODE REVIEW

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libucl${secondaryArchSuffix}_devel = $portVersion
+	devel:libucl${secondaryArchSuffix} = $libVersion
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:29]: No {} needed:
````diff
-	devel:libucl${secondaryArchSuffix} = $libVersion
+	devel:libucl$secondaryArchSuffix = $libVersion
````

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	"
+REQUIRES_devel="
+	libucl${secondaryArchSuffix} == $portVersion base
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:32]: Same here, no {} needed.

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcurl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoreconf
+	cmd:cmake
+	cmd:curl_config$secondaryArchSuffix
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:44]: Is this needed, already provided by the devel:libcurl package?

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
+	"
+
+BUILD()
+{
+	./autogen.sh
+	runConfigure ./configure
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:56]: Suggestion to nuke the static library:
````
	runConfigure ./configure \
		--disable-static
````

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	rm -f $libDir/libucl.a $libDir/libucl.la
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:63]: With the suggestion you should be able to leave out the removal of libucl.a

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+	# set up the develop directory correctly
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:65]: The description here is not fully accurate, what it does is symlinks the libraries to the develop package and moves the libucl.so there, so maybe you can drop this line?

**Review by Begasus — CHANGES_REQUESTED** [review]

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 1 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 2 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 3 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: looks like it is not needed (tried removing and was able to build a package)

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 4 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 5 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 6 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated (removed)

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 7]
```diff
@@ -0,0 +1,75 @@
+SUMMARY=" Universal configuration library parser"
+DESCRIPTION="Libucl is a parser and C API to parse and generate ucl objects."
+HOMEPAGE="https://github.com/vstakhov/libucl"
+COPYRIGHT="2013-2014, Vsevolod Stakhov"
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:4]: No comma in here:

```suggestion
COPYRIGHT="2013-2014 Vsevolod Stakhov"
```

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcurl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoreconf
+	cmd:cmake
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:43]: cmd:cmake?

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:sed
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:49]: cmd:sed not required

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel \
+		$developDir
+}
+
+TEST()
+{
+	make check
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:74]: Missing `cmd:xz` for the tests (you can see this if you run `make VERBOSE=1 check` :)

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 7 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 11]
```diff
@@ -0,0 +1,75 @@
+SUMMARY=" Universal configuration library parser"
+DESCRIPTION="Libucl is a parser and C API to parse and generate ucl objects."
+HOMEPAGE="https://github.com/vstakhov/libucl"
+COPYRIGHT="2013-2014, Vsevolod Stakhov"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/vstakhov/libucl/archive/refs/tags/$portVersion.tar.gz"
+CHECKSUM_SHA256="f63ddee1d7f5217cac4f9cdf72b9c5e8fe43cfe5725db13f1414b0d8a369bbe0"
+SOURCE_FILENAME="$portVersion.tar.gz"
```
[the remark this answers is not in the document]
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 8 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: double checked and this looks unnecessary, removed in update

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 9 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 10 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: thanks, added `cmd:xz`

Comment (Begasus) [reply dev-libs/libucl/libucl-0.9.2.recipe]: cmd:xz isn't required for BUILD_PREREQUIRES, that one should go to TEST_REQUIRES. :)

Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: ah, gotcha.  updated.

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+REVISION="1"
+SOURCE_URI="https://github.com/vstakhov/libucl/archive/refs/tags/$portVersion.tar.gz"
+CHECKSUM_SHA256="f63ddee1d7f5217cac4f9cdf72b9c5e8fe43cfe5725db13f1414b0d8a369bbe0"
+SOURCE_FILENAME="libucl-$portVersion.tar.gz"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="7.2.0"
+libVersionCompat="$libVersion compat >= 7"
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:15]: Missed a few (almost there), thanks for your patience!

```suggestion
libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
```

File: dev-libs/libucl/libucl-0.9.2.recipe [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libucl${secondaryArchSuffix}_devel = $portVersion
+	devel:libucl$secondaryArchSuffix = $libVersion
```
Comment (Begasus) [inline dev-libs/libucl/libucl-0.9.2.recipe:29]:
```suggestion
	devel:libucl$secondaryArchSuffix = $libVersionCompat
```

**Review by Begasus — CHANGES_REQUESTED** [review]

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 12 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

File: dev-libs/libucl/libucl-0.9.2.recipe (hunk 13 above)
Comment (hoanga) [reply dev-libs/libucl/libucl-0.9.2.recipe]: updated

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks! Should be good now. 👍

**Pull request merged**